src/Appender/FileAppender.cs

```diff
@@ -876,6 +886,57 @@ public override void OnClose()
 		}
 #endif
 
+		/// <summary>
+		/// Method for secure open or create Mutex with Synchronize and Modify rights. 
+		/// Using this method allows different users run processes which can open or create a shared mutex without experiencing UnauthorizedAccessException.
+		/// </summary>
+		/// <param name="mutexId">The mutex Id. Here we use a mutex freindly name.</param>
+		/// <param name="errorHandler">Optional error handler, to log potentialy occuring errors.</param>
+		/// <returns></returns>
+		protected static Mutex SecureCreateMutex(string mutexId, IErrorHandler errorHandler)
+		{
+			// Using this mehotd to avoid the issue:
+			// Mutex ~ Access to the path is denied in log4net.Appender.RollingFileAppender.ActivateOptions()
+			// https://jira.apache.org/jira/browse/LOG4NET-587
+			// Solution implemented according to:
+			// UnauthorizedAccessException when trying to open a mutex
+			// https://stackoverflow.com/questions/19536697/unauthorizedaccessexception-when-trying-to-open-a-mutex
+			//NOTE: there are several other jira issues reporting this problem.
+			if (string.IsNullOrWhiteSpace(mutexId))
+			{
+				return null;
```

Comment:
Would you please add an invocation to errorHandler such that even this rare case appears in the internal logs of the framework?

Reply:
yes, I'll do it.
I was thinking to skip this `if (string.IsNullOrWhiteSpace(mutexId))` altogether, or better just skip the `return null;` in this `if`, since if the `mutexId` is null, then the created mutex will be created unnamed. Yes such situation is strange i.m.h.o, and worth of reporting in log, since accessing such ambiguous mutex is error prone. 
I'll make an update to the pull request later today.

Reply:
At best this is error prone. Worst case is that rolling/logging is not synchronized and fails later on with a big bang or silently by wiping the contents of the files.